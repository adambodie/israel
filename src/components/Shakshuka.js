import React, { Component} from 'react';
import ReactMarkdown from 'react-markdown';
import { Link } from 'react-router-dom';
import Modal from 'react-modal';
import { Row, Column } from 'react-foundation';
import { shakshuka } from '../data/articles';
import { mainLink } from '../data/articles';
import { imageLink } from '../data/articles';

const customStyles = {
	content : {
		top: '50%',
		left: '50%',
		right: 'auto',
		bottom: 'auto',
		marginRight: '-50%',
		transform: 'translate(-50%, -50%)'
	}
};

export default class Shakshuka extends Component {
	constructor() {
		super();
		this.state = {
			modalIsOpen: false,
			image: '',
			eggs: [
				{'class': 'eggOne', 'title': 'McDonalds in Israel', 'image': '35'},
				{'class': 'eggTwo',  'title': 'Kosher Mehadrin', 'image': '37'},
				{'class': 'eggThree',  'title': 'Ahuzat Beit Hakerem', 'image': '38'},
				{'class': 'eggFour',  'title': 'Ahuzat Beit Hakerem', 'image': '39'},
				{'class': 'eggFive',  'title': 'Keres Shopping Mall Square', 'image': '47'}
			]
		};
		this.openModal = this.openModal.bind(this);
		this.closeModal = this.closeModal.bind(this);
	}
	openModal(e, index) {
		const { eggs } = this.state;
		const image = eggs[index].image;
		const title = eggs[index].title;
			this.setState({
				modalIsOpen: true,
				image: image,
				title: title
			});
		}
	closeModal() {
		this.setState({modalIsOpen: false});
	}

	render() {
		const { eggs, title, image, modalIsOpen } = this.state;
		return (
			<Row className='display grid-x shakshuka'>
				<Column large={8} offsetOnLarge={2}>
					<h1>How Not to Enjoy Your Shakshuka</h1>
					<img className='fork' src={`${mainLink}/fork.gif`} alt='fork'/>
					<div className='tomato'>
						{eggs.map((x, index) => (
							<div key={index} data-testid={'egg ' + x.class} className={'egg ' + x.class} onClick={() => this.openModal(this, index)}></div>
							))
						}
					</div>
					<p><strong>Click on each egg above to view a picture</strong></p>
					<ReactMarkdown source={shakshuka} className='article' />
					<div className='prev-next'>
						<Link to={ '/yad-vashem' } className='custom-button'>Prev</Link>
						<Link to={ '/herzl' } className='custom-button'>Next</Link>
					</div>
				</Column>
				<Modal
					isOpen={modalIsOpen}
					onRequestClose={this.closeModal}
					style={customStyles}
					contentLabel='Shakshuka Modal'
					appElement={document.getElementById('root')}
				>
					<h2>{title}</h2>
					<img className='thumbnail' src={`${imageLink}/IMG_45${image}.JPG`} alt='title'/>
					<button className='shakshuka-button' onClick={this.closeModal}>x</button>
				</Modal>
		</Row>
		)
	}
}


