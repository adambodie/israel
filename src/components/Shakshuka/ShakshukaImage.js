import React from 'react'

const ShakshukaImage = ({title, image}) => (
	<React.Fragment>
		<h2>{title}</h2>
		<img className='thumbnail' src={`/assets/photos/IMG_45${image}.JPG`} alt={title}/>
	</React.Fragment>
)

export default ShakshukaImage
