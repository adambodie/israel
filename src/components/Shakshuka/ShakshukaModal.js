import React, { Component } from 'react'
import  Modal  from 'react-modal'
import { connectModal } from 'redux-modal'

const SHAKSHUKA_STYLES = {
	content : {
		top: '50%',
		left: '50%',
		right: 'auto',
		bottom: 'auto',
		marginRight: '-50%',
		transform: 'translate(-50%, -50%)'
	}
}
class MyModal extends Component {
	render() {
		const { show, handleHide, title, image } = this.props
		return (
			<Modal 
				isOpen={show} 
				style={SHAKSHUKA_STYLES}
				appElement={document.getElementById('root')}
				>
					<h2>{ title }</h2>
					<button onClick={handleHide} className='shakshuka-button'>x</button>
					<img className='thumbnail' src={`/assets/photos/IMG_45${image}.JPG`} alt={title}/>

			</Modal>
		)
	}
}

export default class ShakshukaModal extends Component {
	render() {
		const { name } = this.props
		const WrappedMyModal = connectModal({ name })(MyModal)
		return <WrappedMyModal />
	}
}