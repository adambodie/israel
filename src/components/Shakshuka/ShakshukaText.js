import React from 'react'
import ReactMarkdown from 'react-markdown'
import PrevNext from '../PrevNext'
import { connect } from 'react-redux'

const mapStateToProps = state => {
	return {
		shakshuka: state.shakshuka,
	}
}

const ShakshukaText = ({ shakshuka }) => (
	<React.Fragment>
		<p><strong>Click on each egg above to view a picture</strong></p>
		<ReactMarkdown source={shakshuka} className='article' />
		<PrevNext prev={'/yad-vashem'} next={'/herzl'} />
	</React.Fragment>
)

export default connect(mapStateToProps) (ShakshukaText)
