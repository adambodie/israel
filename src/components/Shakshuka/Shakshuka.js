import React, { Component} from 'react';
import Modal from 'react-modal';
import { Row, Column } from 'react-foundation';
import ShakshukaImage from './ShakshukaImage';
import ShakshukaText from './ShakshukaText';

const customStyles = {
	content : {
		top: '50%',
		left: '50%',
		right: 'auto',
		bottom: 'auto',
		marginRight: '-50%',
		transform: 'translate(-50%, -50%)'
	}
};

export default class Shakshuka extends Component {
	constructor() {
		super();
		this.state = {
			modalIsOpen: false,
			image: '',
			eggs: [
				{'class': 'eggOne', 'title': 'McDonalds in Israel', 'image': '35'},
				{'class': 'eggTwo',  'title': 'Kosher Mehadrin', 'image': '37'},
				{'class': 'eggThree',  'title': 'Ahuzat Beit Hakerem', 'image': '38'},
				{'class': 'eggFour',  'title': 'Ahuzat Beit Hakerem', 'image': '39'},
				{'class': 'eggFive',  'title': 'Keres Shopping Mall Square', 'image': '47'}
			]
		};
		this.openModal = this.openModal.bind(this);
		this.closeModal = this.closeModal.bind(this);
	}
	openModal(e, index) {
		const { eggs } = this.state;
		const image = eggs[index].image;
		const title = eggs[index].title;
			this.setState({
				modalIsOpen: true,
				image: image,
				title: title
			});
		}
	closeModal() {
		this.setState({modalIsOpen: false});
	}

	render() {
		const { eggs, title, image, modalIsOpen } = this.state;
		return (
			<Row className='display grid-x shakshuka'>
				<Column large={8} offsetOnLarge={2}>
					<h1>How Not to Enjoy Your Shakshuka</h1>
					<div className='tomato'>
						{eggs.map((x, index) => (
							<div key={index} data-testid={'egg ' + x.class} className={'egg ' + x.class} onClick={() => this.openModal(this, index)}></div>
							))
						}
					</div>
					<ShakshukaText />
				</Column>
				<Modal
					isOpen={modalIsOpen}
					onRequestClose={this.closeModal}
					style={customStyles}
					contentLabel='Shakshuka Modal'
					appElement={document.getElementById('root')}
				>
					<ShakshukaImage title={title} image={image} />
					<button className='shakshuka-button' onClick={this.closeModal}>x</button>
				</Modal>
		</Row>
		)
	}
}


