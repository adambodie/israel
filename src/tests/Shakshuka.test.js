import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Shakshuka from '../components/Shakshuka';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<MemoryRouter><Shakshuka /></MemoryRouter>, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const tree = renderer.create(<MemoryRouter><Shakshuka /></MemoryRouter>).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders a `h2`', () => {
	const wrapper = shallow(<Shakshuka />);
	expect(wrapper.find('h2').length).toEqual(1);
 });

