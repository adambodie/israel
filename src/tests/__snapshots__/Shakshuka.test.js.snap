// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row display grid-x shakshuka"
>
  <div
    className="columns large-8 large-offset-2"
  >
    <h1>
      How Not to Enjoy Your Shakshuka
    </h1>
    <div
      className="tomato"
    >
      <div
        className="egg eggOne"
        onClick={[Function]}
      />
      <div
        className="egg eggTwo"
        onClick={[Function]}
      />
      <div
        className="egg eggThree"
        onClick={[Function]}
      />
      <div
        className="egg eggFour"
        onClick={[Function]}
      />
      <div
        className="egg eggFive"
        onClick={[Function]}
      />
    </div>
    <p>
      <strong>
        Click on each egg above to view a picture
      </strong>
    </p>
    <div
      className="article"
    >
      <p>
        Between Yad Vashem and the Herzl Cemetery was a quick lunch break nearby. Within a nice little mall was a few places to eat.  The first one I saw was McDonalds.  I
        '
        ve been there, done that from my trips to Europe.  I still remember being charged Euro cents for ketchup packets in Austria.  Are you kidding me?  So I passed up the chance at McDonalds.  So what would I have for lunch on this date?   
      </p>
      <p>
        I ended up with a group that opted for a more sit down place than anticipated.  I might have gone for a sandwich, and that would have been fine with me, until someone pointed out they had Shakshuka.  I had have no clue what Shakshuka was.  But I certainly like the idea of trying new foods, as long as it doesn
        '
        t contain foods I don
        '
        t like.  So what exactly is Shakshuka?  Shakshuka is a dish of eggs poached in a tomato sauce with chili peppers and onions, along with various spices, commonly including cayenne pepper, cumin, paprika and nutmeg. That was enough to convince me to pique my interest.
      </p>
      <p>
        I also was interested in trying an Iced Coffee.  Our staff told us we had to try the Iced Coffee at some point, that it was absolutely delicious, and nothing like the Iced Coffee back home.  I passed up the opportunity to try some on the first day in the Old City, and I certainly wanted to have some better coffee than the instant coffee that was served at our hotel.  So now was to time to have some Iced Coffee.  So what makes Iced Coffee different?  Instead of being cold, it
        '
        s more slushee at the top, kind of like a blended coffee but not quite completely blended.  The Iced Coffee back home can be ordered as Cold Coffee.
      </p>
      <p>
        That was cool and it came out quickly, but the Shakshuka did not.  And with limited time, we couldn
        '
        t wait around for it.  After a couple times of checking to see when it would be done, our group finally got our Shakshuka
        '
        s and other dishes, and made our dash towards the bus.  I flagged down our staff and warned them beforehand of the quandary, so there would be no chance we
        '
        d be left behind.  So instead of enjoying Shakshuka sitting down, I and a few others ended up having to scarf it down on the bus ride to Mount Herzl and before our tour started.  The meal itself was tasty, but definitely preferred sitting down and not To-Go.  The only thing that could have been worse was for it to explode in my face, but that
        '
        s not part of my story...
      </p>
    </div>
    <div
      className="prev-next"
    >
      <a
        className="custom-button"
        href="/yad-vashem"
        onClick={[Function]}
      >
        Prev
      </a>
      <a
        className="custom-button"
        href="/herzl"
        onClick={[Function]}
      >
        Next
      </a>
    </div>
  </div>
</div>
`;
